### PR TITLE
Add doc on CI for ICU4X

### PR DESCRIPTION
**Pull request by echeran**
@gnrunge 

Should be generally applicable to other projects within unicode-org

### CODE REVIEW

File: docs/process/ci_build.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+* A workflow runs multiple jobs **in parallel**.
+* Workflows are run by Github Actions any time certain trigger events occur. Each workflow configures what trigger events conditions must be met for it to to be invoked (ex: only for Pull Requests; only on merges to the main branch)
+* Jobs within a workflow can be made to run in a particular order by specifying job dependencies. This creates the directed acyclic graph (DAG) relationship of job execution.
+
+## Getting Started
+
+* This [intro page](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/introduction-to-github-actions) shows the following
+  * A Github Action workflow is written in YAML syntax, and must be placed in `.github/workflows`.
+  * A separate tab within the Github UI for a repository shows the history of all Actions runs for the repository.
+  * In the UI, for each workflow run instance, you can see each job of the workflow, and each step within each job.  For a job,you can see the console output (stdout, stderr) for each step with linkable line numbers and a search bar for searching the logs.
```
Comment (gnrunge) [inline docs/process/ci_build.md:23]: space after ',' at '... for a job,you can ...'

File: docs/process/ci_build.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+* This example from an [old, experimental, preliminary ICU4X workflow](https://github.com/unicode-org/icu4x/commit/466eaa3a72635b01cfc1be33e1c899e33147301a) shows a really small task executed in a naive way:
+  * The `on` key config shows that the workflow is triggered only when a commit is made on `master`. Since the repository only allows commits to `master` through the Pull Request process, this workflow is unhelpful for Pull Request authors and reviewers because pull requests originating from any branch not called `master` (all of them) will not trigger the workflow.
+  * There is one job, which is to build and run unit tests
```
Comment (gnrunge) [inline docs/process/ci_build.md:27]: fullstop at end of sentence.

File: docs/process/ci_build.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    - uses: actions/checkout@v2
+    - ...
+```
+  * A job matrix can help decrease wall clock time for multiple independent long-running steps, like benchmarks. [Example](https://github.com/unicode-org/icu4x/blob/master/.github/workflows/build-test.yml#L115-L127):
+```
+  benchmark:
+    strategy:
+      fail-fast: false
+      matrix:
+        component:
```
Comment (gnrunge) [inline docs/process/ci_build.md:55]: 'component:' looks interesting but is introduced out of the blue. Please add a brief description how components are defined.

File: docs/process/ci_build.md (hunk 1 above)
Comment (echeran) [reply docs/process/ci_build.md]: Done.

File: docs/process/ci_build.md (hunk 2 above)
Comment (echeran) [reply docs/process/ci_build.md]: Done.

File: docs/process/ci_build.md (hunk 3 above)
Comment (echeran) [reply docs/process/ci_build.md:64]: Done. I added a small description.

Btw: `component` is merely another example of a parameter in a parameterized job, as configured by the job matrix.

**Review by zbraniecki — APPROVED** [review]

File: docs/process/ci_build.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  * A Github Action workflow is written in YAML syntax, and must be placed in `.github/workflows`.
+  * A separate tab within the Github UI for a repository shows the history of all Actions runs for the repository.
+  * In the UI, for each workflow run instance, you can see each job of the workflow, and each step within each job.  For a job, you can see the console output (stdout, stderr) for each step with linkable line numbers and a search bar for searching the logs.
+
+* This example from an [old, experimental, preliminary ICU4X workflow](https://github.com/unicode-org/icu4x/commit/466eaa3a72635b01cfc1be33e1c899e33147301a) shows a really small task executed in a naive way:
+  * The `on` key config shows that the workflow is triggered only when a commit is made on `master`. Since the repository only allows commits to `master` through the Pull Request process, this workflow is unhelpful for Pull Request authors and reviewers because pull requests originating from any branch not called `master` (all of them) will not trigger the workflow.
```
Comment (gnrunge) [inline docs/process/ci_build.md:26]: I think in this paragraph describes the 'push: branches: [master]' entry in the sample file. It also contains a 'pull_request:branches: [master]' entry, can you say something about the 'pull_request:' entrance? Does creating of a pull request trigger the action, or the merging of a pull request (which is a commit).

File: docs/process/ci_build.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    * The preferred convention by Github Actions is that action repository maintainers use `v1` and `v2` as tags that point to the commits that are tagged with the latest version within that major version. (Ex: `v1` points to `v1.2.3`, `v2` points to `v2.8.9.1`). However, be aware that not all actions adhere to this guideline, and instead only have `vX.Y.Z` tags without a `vX` tag to the latest `vX.*.*`. As an example, this problem was [observed and worked around in the `unicode-org/cldr` repo](https://github.com/unicode-org/cldr/pull/813).
+  * For this simplistic example, the way to improve it to allow a Pull Request to run the same checks after each new commit is pushed, in the same way that checks are run on merges to `master`, is to change `on.pull_request.branches` to [match all origin branch names](https://github.com/unicode-org/icu4x/commit/16ae4611738fbe94b36e17b77aee6cc541c0a171).
```
Comment (gnrunge) [inline docs/process/ci_build.md:31]: Ah, this paragraph talk about the 'on ... pull_request' entry.

File: docs/process/ci_build.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
...ponent` was created. Then, a testing branch [`ci-bench-per-component-test`](https://github.com/echeran/icu4x/pull/13/commits) was created. Feature changes were made on the feature branch, and the `*-test` branch was always rebased on top of the feature branch in order to maintain fork/testing reconfigurations, such as [allowing `master`-only jobs to run on all branches](https://github.com/echeran/icu4x/pull/13/commits/5e83cb4f97a6629b9bfa330aaeb73b3c523f0a8f), making the [API docs / benchmarks copy destination to be on the personal fork](https://github.com/echeran/icu4x/pull/13/commits/61953d9275d3853d96f43ed95e2ff0e9dc5fdd5f), and adding lots of stdout debugging that gets captured in the logs ([example](https://github.com/echeran/icu4x/pull/13/commits/d6168715ac644b785c920b639cd5238e7a52f806)). The `*-test` branch (`ci-bench-per-component-test`) was pushed to the personal fork after each change to trigger workflow test invocations. The feature branch was largely kept locally until the feature work was done, but the important point was to preserve the invariant that the feature branch (`ci-bench-per-component`) was always in a good state to be turned into a PR on the upstream repo.
```
Comment (gnrunge) [inline docs/process/ci_build.md:39]: As for "pushing directly to `master` on your fork repo's settings", after testing one then can clean up the fork's master branch with 'git reset --hard HEAD~n' followed by a 'push -f' to the master branch on GitHub. 'n' being the number of recent commits to revert. However, this looks a bit messy, not sure we should recommend it.

**Review by gnrunge — APPROVED** [review]

**Pull request merged**